sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryExpression.cs

```diff
@@ -63,16 +66,30 @@ internal DataFactoryExpression(string expression, string type)
         {
             if (HasLiteral)
             {
-                if (_literal is Array literalArray)
+                if (_literal == null)
+                    return null;
+
+                if (typeof(T).IsGenericType && typeof(T).GetGenericTypeDefinition() == typeof(IList<>))
                 {
-                    return $"[{string.Join(",", literalArray.OfType<object>().Select(item => item?.ToString()))}]";
+                    var methodInfo = typeof(DataFactoryExpression<>).MakeGenericType(typeof(IList<>)).GetGenericTypeDefinition().MakeGenericType(typeof(T).GenericTypeArguments[0])
+                        .GetMethod(nameof(ToStringList), BindingFlags.Static | BindingFlags.NonPublic)!
+                        .MakeGenericMethod(typeof(T).GenericTypeArguments[0]);
+                    return (string?) methodInfo!.Invoke(null, new object?[] {_literal})!;
```

Comment:
Is it worth caching this `methodInfo`?

Reply:
Just for readability.

Reply:
I wonder if we should use our own serialization here instead?

Reply:
Do you mean get a string representation of the serialized JSON? I'm not sure we want to tie it that closely to a serialization format.

Reply:
Aren't we re-inventing a different way to do the same thing here?  Id prefer to avoid the reflection if possible here.